data/index.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { createStore } from 'redux';
+import { flowRight } from 'lodash';
+
+/**
+ * Module constants
+ */
+const reducers = [];
+const enhancers = [];
+if ( window.__REDUX_DEVTOOLS_EXTENSION__ ) {
+	enhancers.push( window.__REDUX_DEVTOOLS_EXTENSION__() );
+}
+const store = createStore( dynamicReducer, {}, flowRight( enhancers ) );
+
+/**
+ * Reducer function combining the dynamic "reducers" array
+ *
+ * @param {Object} state  Global state
+ * @param {Object} action Action
+ *
+ * @return {Object}       Updated global state
+ */
+function dynamicReducer( state = {}, action ) {
+	let hasChanges = false;
+	const newState = {};
+	reducers.forEach( ( { key, reducer } ) => {
```

Comment:
I imagine this would be a hot path, so might be worth exploring micro-optimizations, e.g. a standard `for` loop instead of `Array#forEach`:

https://jsperf.com/obj-accumulate-looping/1

(Aside: The original implementation looks better served by an `Array#reduce` than `forEach`).